pkg/rancher-desktop/store/snapshots.ts

```diff
@@ -0,0 +1,74 @@
+import { GetterTree } from 'vuex';
+
+import { fetchAPI } from './credentials';
+import { ActionContext, MutationsType } from './ts-helpers';
+
+import { Snapshot } from '@pkg/main/snapshots/types';
+interface SnapshotsState {
+  snapshots: Snapshot[]
+}
+
+export const state: () => SnapshotsState = () => ({ snapshots: [] });
+
+export const mutations: MutationsType<SnapshotsState> = {
+  SET_SNAPSHOTS(state: SnapshotsState, snapshots: Snapshot[]) {
+    state.snapshots = snapshots;
+  },
+};
+
+type SnapshotsActionContext = ActionContext<SnapshotsState>;
+
+export const actions = {
+  async fetch({ commit, rootState }: SnapshotsActionContext) {
+    const response = await fetchAPI('/v1/snapshots', rootState);
+
+    if (!response.ok) {
+      console.log(`fetchSnapshots: failed: status: ${ response.status }:${ response.statusText }`);
+
+      return;
+    }
```

Comment:
Do we need to relay that there was an error in the UI or will this be in a follow-up PR?

Reply:
This is required in a follow-up, we will need to notify errors during creation and restore to the user.